File: Day 29 : Bitwise AND/Solution.js
function bitwiseAnd(N, K) {
    let max = -1;
    for(let i = 1; i <= N; i++){
        for(let j = i+1; j <= N; j++){
            let res = i & j;
            if(res < K && max < res) {
                max = res;
            }
        }
    }
    return max;
}

function main() {
        const firstMultipleInput = readLine().split(' ');

        const count = parseInt(firstMultipleInput[0], 10);

        const lim = parseInt(firstMultipleInput[1], 10);

        const res = bitwiseAnd(count, lim);

        console.log(res);
}

main();

File: Day 29 : Bitwise AND/Solution.rb
def bitwiseAnd(n, k)  
  if (k-1 | k) <= n
    return k-1
  else
    return k-2
  end
end

t = gets.strip.to_i

t.times do |t_itr|
  first_multiple_input = gets.rstrip.split
  count = first_multiple_input[0].to_i
  lim = first_multiple_input[1].to_i
  res = bitwiseAnd count, lim
  puts res
end
